### ISSUE TITLE
Clarify withDimnames=FALSE in the documentation

### ISSUE THREAD

**Issue by SamGG** [Bioconductor/SummarizedExperiment#79]
Can someone clarify this?
https://github.com/Bioconductor/SummarizedExperiment/blob/59caa627c5fdab4fe0b747ce9051c06227dae282/man/SummarizedExperiment-class.Rd#L180-L183

**Comment by hpages** [comment]
In addition to the dimnames that are set on the SummarizedExperiment object itself, the individual assays that are stored in the object can have their own dimnames or not:
```
library(SummarizedExperiment)
a1 <- matrix(runif(15), ncol=3, dimnames=list(letters[1:5], LETTERS[1:3]))
a2 <- matrix(rpois(15, 0.8), ncol=3)
a3 <- matrix(101:115, ncol=3, dimnames=list(NULL, LETTERS[1:3]))
se <- SummarizedExperiment(SimpleList(a1, a2, a3))
```
The dimnames of the SummarizedExperiment object (top-level dimnames):
```
dimnames(se)
# [[1]]
# [1] "a" "b" "c" "d" "e"
#
# [[2]]
# [1] "A" "B" "C"
```
When you extract assays from the object, by default the top-level dimnames are set on them:
```
assay(se, 2)  # this is 'a2', but with the top-level dimnames on it
#   A B C
# a 1 0 4
# b 1 1 1
# c 3 0 2
# d 0 1 2
# e 2 0 2

assay(se, 3)  # this is 'a3', but with the top-level dimnames on it
#     A   B   C
# a 101 106 111
# b 102 107 112
# c 103 108 113
# d 104 109 114
# e 105 110 115
```
However if you use `withDimnames=FALSE` then you'll get the assays _as-is_, i.e. with their original dimnames (this is how they are stored in the SummarizedExperiment object):
```
assay(se, 2, withDimnames=FALSE)  # identical to 'a2'
#      [,1] [,2] [,3]
# [1,]    1    0    4
# [2,]    1    1    1
# [3,]    3    0    2
# [4,]    0    1    2
# [5,]    2    0    2

assay(se, 3, withDimnames=FALSE)  # identical to 'a3'
#        A   B   C
# [1,] 101 106 111
# [2,] 102 107 112
# [3,] 103 108 113
# [4,] 104 109 114
# [5,] 105 110 115

rownames(se) <- strrep(letters[22:26], 3)

dimnames(se)
# [[1]]
# [1] "vvv" "www" "xxx" "yyy" "zzz"
#
# [[2]]
# [1] "A" "B" "C"

assay(se, 1)
#              A          B         C
# vvv 0.31498111 0.02377229 0.9448467
# www 0.98389056 0.33795122 0.5720427
# xxx 0.06080784 0.86596357 0.9112781
# yyy 0.07528625 0.15423982 0.7116326
# zzz 0.53512940 0.36430820 0.5044493

assay(se, 1, withDimnames=FALSE)  # identical to 'a1'
#            A          B         C
# a 0.31498111 0.02377229 0.9448467
# b 0.98389056 0.33795122 0.5720427
# c 0.06080784 0.86596357 0.9112781
# d 0.07528625 0.15423982 0.7116326
# e 0.53512940 0.36430820 0.5044493
```
Hope this helps.

H.

**Comment by SamGG** [comment]
Hi Hervé.
Thanks for your clear explanation.
Could you update the doc as it mentions twice `withDimnames=FALSE`?
Best.

**Comment by hpages** [comment]
> Could you update the doc as it mentions twice withDimnames=FALSE?

Oh I see! Completely missed that. Thanks for the catch.

Fixed now in **SummarizedExperiment** 1.33.2. Also added "Top-level dimnames vs assay-level dimnames" section to the vignette. See commit 9955ec029d7891320b6ad320da8ccb306dccebae
